src/ProductMapper.java

```diff
@@ -0,0 +1,64 @@
+import java.util.List;
+
+public class ProductMapper extends ProductGroup {
```

Comment:
ProductMapper doesn't need to extend ProductGroup, now ProductMapper has the same methods ProductGroup has which can get a bit confusing/unnecessary. Instead ProductMapper.mapProducts can instantiate a new ProductGroup
```
 public ProductGroup mapProducts(List<Row> rowsOfData) {
    ProductGroup productGroup = new ProductGroup();
    for(Row row : rowsOfData){
       if (isLowestOriginYear(row)) {
                productGroup.setLowestOriginYear(row.getOriginYear());
        }
       .....
     return productGroup;
}
```